android/coretest/src/androidTest/java/com/mongodb/stitch/android/core/StitchAuthListenerIntTests.kt

```diff
@@ -0,0 +1,283 @@
+package com.mongodb.stitch.android.core
+
+import android.support.test.runner.AndroidJUnit4
+import com.google.android.gms.tasks.Tasks
+import com.mongodb.stitch.android.core.auth.StitchAuth
+import com.mongodb.stitch.android.core.auth.StitchAuthListener
+import com.mongodb.stitch.android.core.auth.StitchUser
+import com.mongodb.stitch.android.core.auth.providers.userpassword.UserPasswordAuthProviderClient
+import com.mongodb.stitch.android.testutils.BaseStitchAndroidIntTest
+import com.mongodb.stitch.core.admin.authProviders.ProviderConfigs
+import com.mongodb.stitch.core.admin.userRegistrations.sendConfirmation
+import com.mongodb.stitch.core.auth.providers.anonymous.AnonymousCredential
+import com.mongodb.stitch.core.auth.providers.userpassword.UserPasswordCredential
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class StitchAuthListenerIntTests : BaseStitchAndroidIntTest() {
+    @Test
+    fun testOnUserLoggedInDispatched() {
+        val app = createApp()
+
+        addProvider(app.second, ProviderConfigs.Anon)
+        addProvider(app.second, config = ProviderConfigs.Userpass(
+            emailConfirmationUrl = "http://emailConfirmURL.com",
+            resetPasswordUrl = "http://resetPasswordURL.com",
+            confirmEmailSubject = "email subject",
+            resetPasswordSubject = "password subject")
+        )
+
+        val client = getAppClient(app.first)
+
+        val countDownLatch = CountDownLatch(1)
+
+        client.auth.addAuthListener(object : StitchAuthListener {
+            override fun onAuthEvent(auth: StitchAuth?) {
```

Comment:
[q] can we make `onAuthEvent` default so people don't have to implement it? shouldn't be a breaking change right?